### DISCUSSION TITLE
How can I use upload.single() with if conditions ?

### DISCUSSION THREAD

**Discussion by mrcihantoker** [expressjs/multer#1207] [Q&A]
I am trying to code a "add product" page for my e-commerce website.I want system to get the input values and check the database for the product with the same name.If there isn't a product with the same name;I want system add the photo to 'Image' folder.How can I do it ?(enctype is set to multipart/form-data)

                app.post('/Add-Product-Check',function(req, res) {
                    var product_name = req.body.add_product_product_name;
                    var product_price = req.body.add_product_product_price;
                    var product_category = req.body.add_product_categorie_section;

                    var sql="SELECT COUNT(*) FROM products WHERE Product_name='productname'".replace("productname",product_name);

                    server.query(sql,function(error,product_count_query,fields){

                        if(err){

                            throw err;
                        }

                        else{

                            var product_count=product_count_query[0]['COUNT(*)'];

                            if(product_count>0){

                                console.log("There is a product with this name.");
                            }

                            else{

                                upload.single('add_product_product_photo')(req,res,function(err){

                                    if(err){

                                        throw err;
                                    }

                                    else{

                                        console.log("Photo uploaded!");
                                    }
                                });
                            }
                        }
                    });

                        
                });

**Comment by mrdulin** [comment]
Does your code not work? Is there any error? Show the debugging details.
